### PR TITLE
Optional Curios API integration

### PR DESCRIPTION
**Pull request by cody-greene**
Add *optional* support for [Curios](https://github.com/TheIllusiveC4/Curios) by allowing backpacks to be equipped in a "back" slot instead of the "chest" slot.

This takes the form of a static class which wraps the Curios API and makes sure the mod is loaded before trying to use it. In places where this project checks for an equipped backpack — rendering and opening the inventory — we now check the curio slot first. Finally, if the the curio API is found, then backpacks are no long auto-equipped to the "chest" slot, but go to the "back" slot instead.

### CODE REVIEW

File: src/main/java/com/terraformersmc/campanion/mixin/MixinMobEntity.java [hunk 1]
```diff
@@ -16,7 +17,7 @@
 	@Inject(method = "getPreferredEquipmentSlot", at = @At("HEAD"), cancellable = true)
 	private static void onGetPreferredEquipmentSlot(ItemStack stack, CallbackInfoReturnable<EquipmentSlot> info) {
 		Item item = stack.getItem();
-		if (item instanceof BackpackItem) {
+		if (!Curios.isEnabled() && item instanceof BackpackItem) {
```
Comment (NeusFear) [inline src/main/java/com/terraformersmc/campanion/mixin/MixinMobEntity.java:20]: Is there a way to set the preferred slot to the back slot if curious is enabled?

**Review by NeusFear — COMMENTED** [review]
I'm not 100% sure if we want to merge this or not but if we did here are a few notes. We can discuss wheather we want this merged otherwise.

File: src/main/java/com/terraformersmc/campanion/mixin/MixinMobEntity.java (hunk 1 above)
Comment (cody-greene) [reply src/main/java/com/terraformersmc/campanion/mixin/MixinMobEntity.java:20]: If the Curios UI is **open**, then backpacks will auto-equip to the back slot. The onGetPreferredEquipmentSlot is not needed.

If the Curios UI is **not open**, then backpacks will not auto-equip. I'll look at how I can fix that.

**Comment by superon** [comment]
any update here? i feel like "rendering and opening the inventory" is the important thing here, players/pack owners can create slots and enable items to be placed into them within the curios mod itself so i don't think "allowing backpacks to be equipped in a "back" slot" is super important...   i've been able to create a slot and place a pack in there, but just unable to open it from there.

**Comment by NebelNidas** [comment]
Well, since Curios is discontinued, I think this PR can be closed now. However, I'm still in favour of adding a [Trinkets](https://www.curseforge.com/minecraft/mc-mods/trinkets-fabric) integration (the successor of Curios), see https://github.com/TerraformersMC/Campanion/issues/32!

**Pull request closed without merging**